src/screens/VaultScreen.js

```diff
@@ -13,11 +14,305 @@ export default class VaultScreen extends React.Component {
+      ],
+      fontLoaded: false,
+      resources: [
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 1,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 2,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 3,
+        }
+      ]
+    };
+  }
+  updateFilter(item_id, new_state) {
+    var names_arr = this.state.names.slice()
+    for(var i = 0; i < names_arr.length; i++) {
+      if(names_arr[i].id == item_id) {
+        names_arr[i].selected = new_state;
+      }
+    };
+    this.setState({names:names_arr});
+  }
+  setOpposite(item_id) {
+    for(var i = 0; i < this.state.names.length; i++) {
+      if(this.state.names[i].id == item_id) {
+        this.updateFilter(item_id, !this.state.names[i].selected);
+      }
+    };
+  }
+  falseState(name, item_id) {
+    if(name=='Clear') {
+      for(var i = 1; i < this.state.names.length; i++) {
+        this.updateFilter(this.state.names[i].id, false)
+      }
+    } else {
+      this.setOpposite(item_id);
+    }
+  }
+  filterScroller() {
+    return this.state.names.map((item) => {
+      if (item.selected==false) {
+        return (
+          <TouchableHighlight key = {item.id} style={styles.item} onPress={() => {this.falseState(item.name, item.id);}}>
+            <Text style={{color:'white', fontSize:12, fontFamily: 'source-sans-pro-semibold',}}>{item.name}</Text>
+          </TouchableHighlight>
+        );
+      } else {
+        return (
+          <TouchableHighlight key = {item.id} style = {[styles.item, {backgroundColor:'white',}]} onPress={() => {this.setOpposite(item.id);}}>
+            <Text style={{color:'black', fontSize:12, fontFamily: 'source-sans-pro-semibold',}}>{item.name}</Text>
+          </TouchableHighlight>
+        );
+      }
+    })
+  }
+  upvote(item_id) {
+    var resources_arr = this.state.resources.slice()
+    for(var i = 0; i < resources_arr.length; i++) {
+      if(resources_arr[i].id == item_id) {
+        resources_arr[i].upvotes = resources_arr[i].upvotes+1;
+      }
+    };
```

Comment:
wondering if there is a way to do this by getting the resource by id without looping through the entire resources_arr @lbkchen 

perhaps we will have a better idea once we are getting the data